server/src/main/java/org/opensearch/search/sandbox/SandboxedRequestTrackerService.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.sandbox;
+
+import org.opensearch.core.tasks.resourcetracker.TaskResourceUsage;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskCancellation;
+import org.opensearch.tasks.TaskManager;
+import org.opensearch.tasks.TaskResourceTrackingService;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.LongSupplier;
+import java.util.stream.Collectors;
+
+/**
+ * This class tracks requests per sandboxes
+ */
+public class SandboxedRequestTrackerService implements TaskManager.TaskEventListeners, SandboxResourceTracker, SandboxStatsReporter {
+
+    private static final String CPU = "CPU";
+    private static final String JVM_ALLOCATIONS = "JVM_Allocations";
+    private static final int numberOfAvailableProcessors = Runtime.getRuntime().availableProcessors();
```

Comment:
What happens when a cluster has its nodes adjusted? Let's say you set two requests 

A (10% allocation) and B (20 % allocation) 

You have 10 nodes in your cluster, we could imagine a case where it is made of 4 manager nodes, and 6 normal nodes. Then lets say you have 2 data nodes with 2 replica shards each. You would then have 3 nodes with the same data at any time. 

Now it should be possible that request A gets to use one of the three nodes completely and request B gets the other two nodes (1/10 = 10%, 2/10=20%). But then a node drops meaning A is now using 1/9 and B is using 1/9? 

Now what happens?

Reply:
I guess you mean 2 indices with 1 shard with 2 replica shards for each primary. 

I didn't get your question here, This feature will enforce the resources at a node level not cluster level. 

Can you clarify your question ?